### PR TITLE
EHN: Adding square in paddle.tensor.tesnor.Tensor

### PR DESCRIPTION
**Pull request by MuhammadNizamani**
Adding square in paddle.tensor.tesnor.Tensor from issue #16733

### CODE REVIEW

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 1]
```diff
@@ -159,3 +159,10 @@ def tanh(self, name=None):
     @with_supported_dtypes({"2.4.3 and above": ("float32", "float64")}, "paddle")
     def add_(self, name=None):
         return ivy.add(self._ivy_array)
+
+    with_supported_dtypes(
+        {"2.4.2 and below": ("float16", "float32", "float64")}, "paddle"
```
Comment (juliagsy) [inline ivy/functional/frontends/paddle/tensor/tensor.py:164]: Hey! seems like `float16` is actually unsupported here, would you mind to remove it from the list? Thanks!

Comment (MuhammadNizamani) [reply ivy/functional/frontends/paddle/tensor/tensor.py:164]: I checked paddle docs and it says it can take float16 as input.

Comment (juliagsy) [reply ivy/functional/frontends/paddle/tensor/tensor.py:164]: yeah, but that might be for the functional only as I tried the method out and it gave me errors

Comment (MuhammadNizamani) [reply ivy/functional/frontends/paddle/tensor/tensor.py:164]: Ok I am going to change it thanks 👍

Comment (MuhammadNizamani) [reply ivy/functional/frontends/paddle/tensor/tensor.py]: I have removed float16.

**Comment by MuhammadNizamani** [comment]
@juliagsy  why these checks falied?

**Comment by juliagsy** [comment]
Thanks for contributing!

**Pull request merged**